server/graphql/core/src/loader/sync_file_reference.rs

```diff
@@ -0,0 +1,230 @@
+use repository::{
+    sync_file_reference::{SyncFileReference, SyncFileReferenceFilter, SyncFileReferenceSort},
+    Pagination, RepositoryError, StorageConnectionManager, SyncFileReferenceSortField,
+};
+use repository::{EqualFilter, SyncFileReferenceRepository};
+
+use async_graphql::dataloader::*;
+use async_graphql::*;
+use std::collections::HashMap;
+
+pub struct SyncFileReferenceLoader {
+    pub connection_manager: StorageConnectionManager,
+}
+
+#[async_trait::async_trait]
+impl Loader<String> for SyncFileReferenceLoader {
+    type Value = Vec<SyncFileReference>;
+    type Error = RepositoryError;
+
+    async fn load(&self, ids: &[String]) -> Result<HashMap<String, Self::Value>, Self::Error> {
+        let connection = self.connection_manager.connection()?;
+        let repo = SyncFileReferenceRepository::new(&connection);
+
+        let sync_file_references = repo.query(
+            Pagination::all(),
+            Some(
+                SyncFileReferenceFilter::new()
+                    .record_id(EqualFilter::equal_any(ids.to_owned()))
+                    .is_deleted(false),
+            ),
+            Some(SyncFileReferenceSort {
+                key: SyncFileReferenceSortField::FileName,
+                desc: Some(false),
+            }),
+        )?;
+
+        let mut map: HashMap<String, Vec<SyncFileReference>> = HashMap::new();
+
+        for sync_file_reference in sync_file_references {
+            let asset_id = sync_file_reference
```

Comment:
```suggestion
            let file_id = sync_file_reference
```

Reply:
but it's returning the `record_id` which in this case is the asset id - the `file_id` would be the `id` on the sync_file_reference table wouldn't it?

Reply:
Yes you are right sorry must have miss read the code.